tempodb/tempodb.go

```diff
@@ -382,6 +382,11 @@ func (rw *readerWriter) pollBlocklist() {
 		if err != nil {
 			metricBlocklistErrors.WithLabelValues(tenantID).Inc()
 			level.Error(rw.logger).Log("msg", "error polling blocklist", "tenantID", tenantID, "err", err)
+			rw.blockListsMtx.Lock()
+			delete(rw.blockLists, tenantID)
+			delete(rw.compactedBlockLists, tenantID)
+			rw.blockListsMtx.Unlock()
```

Comment:
Should this be part of the `err != nil` check? The call could error out for a number of reasons. And a failed call should not end up emptying the in-memory blocklist. Rather, this should be under a check `len(blockIDs) == 0`.